### PR TITLE
Add OTP 21.2.3 images

### PR DESCRIPTION
**Pull request by dincho**
- update rebar3 to 3.9.0

### CODE REVIEW

**Comment by lucafavatella** [comment]
From https://circleci.com/gh/aeternity/docker-builder/59
```
#!/bin/bash -eo pipefail
docker build -t $DOCKERHUB_REPO:$TAG \
  --build-arg BUILD_OTP_VERSION=${BUILD_OTP_VERSION:-20.3.8} $DOCKERFILE_PATH
docker login -u $DOCKER_USER -p $DOCKER_PASS
docker push $DOCKERHUB_REPO:$TAG
...
8badb0701e5f: Pushing  517.5MB
$<3>$<3>
24542a28421d: Retrying in 7 seconds 
$<3>$<3>
8badb0701e5f: Pushing  520.2MB
$<3>$<3>
8badb0701e5f: Pushing  522.4MB
$<3>$<3>
8badb0701e5f: Pushing  522.9MB
$<3>$<3>
24542a28421d: Retrying in 6 seconds 
$<3>$<3>
24542a28421d: Retrying in 5 seconds 
$<3>$<3>
24542a28421d: Retrying in 4 seconds 
$<3>$<3>
8badb0701e5f: Pushed 
$<3>$<3>
24542a28421d: Retrying in 3 seconds 
$<3>$<3>
24542a28421d: Retrying in 2 seconds 
$<3>$<3>
24542a28421d: Retrying in 1 second 
$<3>$<3>
24542a28421d: Pushing  4.096kB/335.3kB
$<3>$<3>
24542a28421d: Pushing  352.3kB
$<3>$<3>
24542a28421d: Pushed 
received unexpected HTTP status: 500 Internal Server Error
Exited with code 1
```

File: Dockerfile [hunk 1]
```diff
@@ -24,10 +25,10 @@ RUN LIBSODIUM_DOWNLOAD_URL="https://github.com/jedisct1/libsodium/releases/downl
     && cd libsodium-src \
     && ./configure && make -j$(nproc) && make install && ldconfig
 
-ENV REBAR3_VERSION="3.5.0"
+ENV REBAR3_VERSION="3.9.0"
 RUN set -xe \
     && REBAR3_DOWNLOAD_URL="https://github.com/erlang/rebar3/archive/${REBAR3_VERSION}.tar.gz" \
-    && REBAR3_DOWNLOAD_SHA256="e95e9d1f2ce219f548d4f49ad41409af02069190f19e2b6717585eef6ee77501" \
+    && REBAR3_DOWNLOAD_SHA256="9ea73ce4e60ad4b3108641eae73b4098fadb510142e672ad8e3a793f57e9f992" \
```
Comment (michalbagrowski) [inline Dockerfile]: should be this passed as env variable? i think we do not have anything else hardcoded like that.

File: Dockerfile [hunk 2]
```diff
@@ -24,10 +25,10 @@ RUN LIBSODIUM_DOWNLOAD_URL="https://github.com/jedisct1/libsodium/releases/downl
     && cd libsodium-src \
     && ./configure && make -j$(nproc) && make install && ldconfig
 
-ENV REBAR3_VERSION="3.5.0"
+ENV REBAR3_VERSION="3.9.0"
```
Comment (valkov-ivaylo) [inline Dockerfile]: How does this affect 18.04, where according to [build](../../../aeternity/blob/master/docs/build.md) instructions the only thing that must be installed is Erlang from Ubuntu repositories?

**Comment by lucafavatella** [comment]
> From https://circleci.com/gh/aeternity/docker-builder/59
> 
> ```
> ...
> received unexpected HTTP status: 500 Internal Server Error
> Exited with code 1
> ```

I took the liberty of restarting CI. (I saw a few CI failure on Fri.)

File: Dockerfile (hunk 2 above)
Comment (lucafavatella) [reply Dockerfile]: It does not affect [those build instructions](https://github.com/aeternity/aeternity/blob/831e1ee30e4092058ca35573bdfd8dd5f392c01d/docs/build.md), as those use `rebar3` binary in that repo:
* https://github.com/aeternity/aeternity/blob/831e1ee30e4092058ca35573bdfd8dd5f392c01d/.dockerignore#L5
* https://github.com/aeternity/aeternity/blob/831e1ee30e4092058ca35573bdfd8dd5f392c01d/Dockerfile#L4
* https://github.com/aeternity/aeternity/blob/831e1ee30e4092058ca35573bdfd8dd5f392c01d/Makefile#L4

Also, I see this image is [targeted specifically for CI](https://github.com/aeternity/docker-builder/blame/60305f9268463bb46edded55a3bed1d7d5a784a4/README.md#L3) - not users. I see rebar3 was added specifically for [one such repo's CI](https://github.com/aeternity/aesophia/pull/20).

Comment (lucafavatella) [reply Dockerfile]: I saw rebar 3.5.0 embedded in `package_ubuntu_18.04/Dockerfile` - blocking until I understand that.

Comment (lucafavatella) [reply Dockerfile]: I believe this shall be addressed by https://github.com/aeternity/docker-builder/pull/15

File: Dockerfile (hunk 1 above)
Comment (dincho) [reply Dockerfile]: You can't change the rebar version with build args, thus the checksum is hardcoded here.

**Pull request merged**